dev/archery/archery/integration/datagen.py

```diff
@@ -193,6 +193,28 @@ def generate_range(self, size, lower, upper, name=None,
         return PrimitiveColumn(name, size, is_valid, values)
 
 
+# Integer field that fulfils the requirements for the run ends field of REE.
+# The integers are positive and in a strictly increasing sequence
+class RunEndsField(IntegerField):
+    def __init__(self, name, bit_width, *, nullable=False,
+                 metadata=None):
+        super().__init__(name, is_signed=True, bit_width=bit_width,
+                         nullable=nullable, metadata=metadata, min_value=1)
+
+    def generate_range(self, size, lower, upper, name=None,
+                       include_extremes=False):
+        rng = np.random.default_rng()
+        values = rng.choice(2 ** (self.bit_width - 1) - 1, size=size, replace=False)
```

Comment:
Would be worth adding a comment to explain the strategy here.

Reply:
added